comp/api/api/apiimpl/observability/telemetry.go

```diff
@@ -22,30 +22,46 @@ const (
 	metricHelp      = "Request duration distribution by server, method, path, and status (in seconds)."
 )
 
-// TelemetryHandler is a middleware which sends a metric representing the duration of the request,
-// tagged with the given server name, response status code, path, and method.
-func TelemetryHandler(telemetry telemetry.Component, serverName string) mux.MiddlewareFunc {
-	return telemetryHandler(telemetry, serverName, clock.New())
+type telemetryMiddlewareFactory struct {
+	requestDuration telemetry.Histogram
+	clock           clock.Clock
 }
 
-func telemetryHandler(telemetry telemetry.Component, serverName string, clock clock.Clock) mux.MiddlewareFunc {
-	tags := []string{"servername", "status_code", "method", "path"}
-	var buckets []float64 // use default buckets
-	requestDuration := telemetry.NewHistogram(metricSubsystem, metricName, tags, metricHelp, buckets)
+// TelemetryMiddlewareFactory creates a telemetry middleware tagged with a given server name
+type TelemetryMiddlewareFactory interface {
+	Middleware(serverName string) mux.MiddlewareFunc
+}
 
+func (th *telemetryMiddlewareFactory) Middleware(serverName string) mux.MiddlewareFunc {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			var code int
 			next = extractStatusHandler(&code)(next)
 
 			var duration time.Duration
-			next = timeHandler(clock, &duration)(next)
+			next = timeHandler(th.clock, &duration)(next)
 
 			next.ServeHTTP(w, r)
 
 			path := extractPath(r)
-			metricWithTags := requestDuration.WithValues(serverName, strconv.Itoa(code), r.Method, path)
+			metricWithTags := th.requestDuration.WithValues(serverName, strconv.Itoa(code), r.Method, path)
 			metricWithTags.Observe(duration.Seconds())
 		})
 	}
 }
+
+func newTelemetryMiddlewareFactory(telemetry telemetry.Component, clock clock.Clock) TelemetryMiddlewareFactory {
+	tags := []string{"servername", "status_code", "method", "path"}
+	var buckets []float64 // use default buckets
```

Comment:
Do we have to create variables for this? It seems that is not reuse anywhere else.

Reply:
For `buckets` it's just to make it clear that the `nil` value means using default buckets (it's not clear in the component itself, I had to look into prometheus to figure it out...)
For `tags` it's just to avoid a too long line 😄